### PR TITLE
Fix:  fix for issue in to-match-string.

### PR DESCRIPTION
**Pull request by PaulJPhilp**
Expanded the API for to-match-string test API to address the issues with
matching strings with lookarounds.

interface MatchTypeOptions {
  exactString: boolean;
  substring?: string;
}

export function toMatchString(
  this: jest.MatcherContext,
  received: RegExp | RegexSequence,
  expected: string,
  matchType?: MatchTypeOptions,
) {

If the matchType parameter is not present, toMatchString behaves as previous.

If the matchType parameter is present and the exactString flag is true, toMatchString returns true
only if the exact string is matched.  No substrings.

If the matchType parameter is present and the exactString flag is false, toMatchString returns true
only if the substring is a matched substring of the passed string.

The following test files have been updated to use this new API:

example-currency.ts
example-filename.ts
example-suffixes.ts
example-findwords.ts

### CODE REVIEW

File: test-utils/to-match-string.ts [hunk 1]
```diff
@@ -1,24 +1,40 @@
...
 export function toMatchString(
   this: jest.MatcherContext,
   received: RegExp | RegexSequence,
   expected: string,
+  matchType?: MatchTypeOptions,
 ) {
   const receivedRegex = wrapRegExp(received);
   const matchResult = expected.match(receivedRegex);
+
+  let pass: boolean = false;
```
Comment (mdjastrzebski) [inline test-utils/to-match-string.ts:18]: I have trouble understanding what is the purpose of each of additional options. They seem not to be of any standard JS regex options for `String.match`, `RegExp.exec`, etc. Please explain their purpose as code comments.

File: test-utils/to-match-string.ts [hunk 2]
```diff
@@ -1,24 +1,40 @@
 import type { RegexSequence } from '../src/types';
 import { wrapRegExp } from './utils';
 
+interface MatchTypeOptions {
+  exactString: boolean;
```
Comment (mdjastrzebski) [inline test-utils/to-match-string.ts:5]: Why is this option mandatory? It generate a lot of noise in the testing code. If possible pick one of the values `true` or `false` as default.

File: src/__tests__/example-currency.ts [hunk 3]
```diff
@@ -22,18 +22,21 @@ test('example: extracting currency values', () => {
     endOfString,
   ]);
 
-  expect(currencyRegex).toMatchString('$10');
-  expect(currencyRegex).toMatchString('$ 10');
+  expect(currencyRegex).toMatchString('$10', { exactString: false, substring: '10' });
+  expect(currencyRegex).toMatchString('$ 10', { exactString: false, substring: '10' });
   expect(currencyRegex).not.toMatchString('$ 10.');
-  expect(currencyRegex).toMatchString('$ 10');
+  expect(currencyRegex).toMatchString('$ 10', { exactString: false, substring: '10' });
```
Comment (mdjastrzebski) [inline src/__tests__/example-currency.ts:28]: It looks like this test could be expressed as existing `toMatchGroups` helper:
```ts
expect(currencyRegex).toMatchGroups('$ 10', ['10']);
```

**Comment by PaulJPhilp** [comment]
True.  Even though there are no groups in the pattern.  This is not very
intuitive, at all.

**Comment by PaulJPhilp** [comment]
I have wasted well over a week of my life on bugs in this library.  The author of the library
has not helped in resolving these issues at all.

I will wait for a stable release and evaluate the library again at that point.

**Pull request closed without merging**